__tests__/app.test.js

```diff
@@ -0,0 +1,53 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const data = require("../db/data/test-data");
+const endpoints = require("../endpoints.json");
+
+beforeAll(() => seed(data));
+afterAll(() => db.end());
+
+describe("Topics API", () => {
+	describe("Get Request", () => {
+		test("Get: 200, should return all topics", () => {
+			return request(app)
+				.get("/api/topics")
+				.expect(200)
+				.then(({ body }) => {
+					const { topics } = body;
+					expect(topics).toBeInstanceOf(Array);
+					expect(topics.length).not.toBeLessThan(1);
+					topics.forEach((topic) => {
+						expect(Object.keys(topic)).toEqual(["slug", "description"]);
+					});
+				});
+		});
+	});
+});
+describe("Endpoints API", () => {
+	describe("Get Request", () => {
+		test("endpoints.json formatting correct", () => {
+			expect(endpoints).toBeInstanceOf(Object);
+		});
+		test("Get: 200, should return a list of endpoints from json file", () => {
+			return request(app)
+				.get("/api")
+				.expect(200)
+				.then(({ body }) => {
+					const endpointList = body.endpoints;
+					expect(endpointList).toEqual(endpoints);
+					expect(endpointList).not.toBe(endpoints);
+				});
+		});
+	});
```

Comment:
This could just be 1 full integration test rather than checking it's an object and then the contents after. Requiring in the JSON file and asserting its contents are equal to the response body is absolutely fine to do, well done.

Reply:
I was thinking that checking the json contents against the response would allow me to keep using this test once I start adding to the json file in the future.